crates/orderbook/src/cow_subsidy.rs

```diff
@@ -24,67 +24,114 @@ impl Default for FixedCowSubsidy {
 
 #[async_trait::async_trait]
 impl CowSubsidy for FixedCowSubsidy {
+    /// On success returns a factor of how much of the usual fee an account has to pay based on the
+    /// amount of COW tokens it owns.
     async fn cow_subsidy_factor(&self, _: H160) -> Result<f64> {
         Ok(self.0)
     }
 }
 
-#[derive(Debug, Clone, Copy)]
-enum SubsidyTier {
-    None,
-    Full,
+#[derive(Copy, Clone, Debug)]
+pub struct SubsidyTier {
+    /// How many base units of COW someone needs to own to qualify for this subsidy tier.
+    threshold: U256,
+    /// How much of the usual fee a user in this tier has to pay.
+    fee_factor: f64,
+}
+
+impl std::str::FromStr for SubsidyTier {
+    type Err = anyhow::Error;
+    fn from_str(tier: &str) -> Result<Self, Self::Err> {
+        let mut parts = tier.split(':');
+        let threshold = parts.next().ok_or(anyhow::anyhow!("missing threshold"))?;
+        let threshold: f64 = threshold
+            .parse()
+            .map_err(|_| anyhow::anyhow!("can not parse threshold {} as f64", threshold))?;
+
+        let fee_factor = parts.next().ok_or(anyhow::anyhow!("missing fee factor"))?;
+        let fee_factor: f64 = fee_factor
+            .parse()
+            .map_err(|_| anyhow::anyhow!("can not parse fee factor {} as f64", fee_factor))?;
+
+        if parts.next().is_some() {
+            anyhow::bail!("too many arguments for subsidy tier");
+        }
```

Comment:
I recently discovered [`split_once`](https://doc.rust-lang.org/std/primitive.str.html#method.split_once) when doing this year's Advent of Code which I think would be quite helpful here.